enaml/core/import_hooks.py

```diff
@@ -368,26 +423,203 @@ def get_code(self):
             path to the module as a string.
 
         """
-        # If the .enaml file does not exists, just use the .enamlc file.
-        # We can presume that the latter exists because it was already
-        # checked by the loader. Should the situation ever arise that
-        # it was deleted between then and now, an IOError is more
-        # informative than an ImportError.
         file_info = self.file_info
-        if not os.path.exists(file_info.src_path):
-            code = self._load_cache(file_info)
-            return (code, file_info.src_path)
 
-        # Use the cached file if it exists and is current
-        src_mod_time = int(os.path.getmtime(file_info.src_path))
-        if os.path.exists(file_info.cache_path):
-            magic, ts = self._get_magic_info(file_info)
-            if magic == MAGIC and src_mod_time <= ts:
-                code = self._load_cache(file_info, set_src=True)
-                return (code, file_info.src_path)
+        # Try to load from the cache
+        cache = self.get_cached_code(file_info.src_path)
+        if cache is not None:
+            return cache
 
         # Otherwise, compile from source and attempt to cache
-        return self.compile_code()
+        return self.compile_code(file_info.src_path)
+
+
+#------------------------------------------------------------------------------
+# Enaml Zip Importer
+#------------------------------------------------------------------------------
+class EnamlZipImporter(EnamlImporter):
+
+    #: For potential future changes.
+    supported_archives = {
+        '.zip': ZipFile,
+    }
+
+    @classmethod
+    def locate_module(cls, fullname, path=None):
+        """ Searches for the given Enaml module within a zip and returns an instance
+        of this class on success.  
+        
+        Caching
```

Comment:
Caching is not recognized section in Numpydoc and will break the docs. You can simply omit the section.